File: snippets/fsharp/System/Guid/CompareTo/compareto2.fs
module compareto2

// <Snippet1>
open System

type Comparison =
    | ``Less Than`` = -1
    | Equals = 0
    | ``Greater Than`` = 1

let mainGuid = 
    Guid.Parse "01e75c83-c6f5-4192-b57e-7427cec5560d"

let guid2 = Guid(0x01e75c83, 0xc6f5s, 0x4192s, [| 0xb5uy; 0x7euy; 0x74uy; 0x27uy; 0xceuy; 0xc5uy; 0x56uy; 0x0cuy |])
let guid3 = 
    Guid.Parse("01e75c84-c6f5-4192-b57e-7427cec5560d")

printfn $"{mainGuid} {mainGuid.CompareTo guid2 |> enum<Comparison> :F} {guid2}"
printfn $"{mainGuid} {mainGuid.CompareTo guid3 |> enum<Comparison> :F} {guid3}"

// The example displays the following output:
//    01e75c83-c6f5-4192-b57e-7427cec5560d Greater Than 01e75c83-c6f5-4192-b57e-7427cec5560c
//    01e75c83-c6f5-4192-b57e-7427cec5560d Less Than 01e75c84-c6f5-4192-b57e-7427cec5560d
// </Snippet1>

File: snippets/csharp/System/Guid/CompareTo/compareto2.cs
// <Snippet1>
using System;

public class GuidCompareToExample2
{
    public static void Run()
    {
        Guid mainGuid = Guid.Parse("01e75c83-c6f5-4192-b57e-7427cec5560d");
        unchecked
        {
            Guid guid2 = new(0x01e75c83, (short)0xc6f5,
                                  0x4192,
                                  new byte[] { 0xb5, 0x7e, 0x74, 0x27, 0xce, 0xc5, 0x56, 0x0c });
            Guid guid3 = Guid.Parse("01e75c84-c6f5-4192-b57e-7427cec5560d");

            Console.WriteLine($"{mainGuid} {(Comparison)mainGuid.CompareTo(guid2):F} {guid2}");
            Console.WriteLine($"{mainGuid} {(Comparison)mainGuid.CompareTo(guid3):F} {guid3}");
        }
    }

    private enum Comparison
    { LessThan = -1, Equals = 0, GreaterThan = 1 }
}
// The example displays the following output:
//    01e75c83-c6f5-4192-b57e-7427cec5560d GreaterThan 01e75c83-c6f5-4192-b57e-7427cec5560c
//    01e75c83-c6f5-4192-b57e-7427cec5560d LessThan 01e75c84-c6f5-4192-b57e-7427cec5560d
// </Snippet1>
